app/controllers/application_controller.rb

```diff
@@ -22,12 +23,11 @@ def append_to_log(params)
   # Rails' instrumentation code, and is run after each request.
   def append_info_to_payload(payload)
     super
-    if @custom_log_items
-      payload[:custom_log_items] = @custom_log_items
-    end
-  end
 
-  # :nocov:
+    append_to_log(request_id: RequestStore.store[:request_id])
```

Comment:
Whatever, that name seems fine.

I'm not keen on overuse of before filters but in this case it seems entirely fine: it is genuinely something that should happen before each action, and conceptually makes sense to be the controller's responsibility. For me a middleware would be less clear.